### PR TITLE
lxd/vm: Move bus allocator to own file

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

**Comment by stgraber** [comment]
The logic remains identical so this should not have any visible runtime impact.

**Comment by stgraber** [comment]
@tomponline this should mostly match what we discussed.

File: lxd/instance/drivers/driver_qemu_bus.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		}
+	}
+
+	return a.rootPort
+}
+
+// allocate() does any needed port allocation and  returns the bus name,
+// address and whether the device needs to be configured as multi-function.
+//
+// The group parameter allows for grouping devices together as a single
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu_bus.go:50]: @stgraber This part "The group parameter allows for grouping devices together as a single multi-function device. It automatically keeps track of the number of functions already used and will allocate a new device as needed." seems to conflict, as it initially suggests providing a non-empty `group` argument will cause it to allocate ports using a single multifunction device, but then suggests that additional devices could be allocated.

Comment (tomponline) [reply lxd/instance/drivers/driver_qemu_bus.go:50]: Perhaps rewording to "The group parameter allows for grouping devices together on one or more multi-function ports. It automatically keeps track of the number of functions already used and will allocate new multi-function ports as needed"

**Comment by tomponline** [comment]
@stgraber looks good thanks, only a minor comment suggestion, so as its been merged anyway I'll send as a separate PR.

**Pull request merged**